molr-mole-core/src/main/java/io/molr/mole/core/runnable/lang/ctx/OngoingContextualBranch.java

```diff
@@ -0,0 +1,31 @@
+package io.molr.mole.core.runnable.lang.ctx;
+
+import io.molr.commons.domain.Block;
+import io.molr.mole.core.runnable.RunnableLeafsMission;
+import io.molr.mole.core.runnable.lang.BranchMode;
+import io.molr.mole.core.runnable.lang.GenericOngoingBranch;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+
+import static java.util.Objects.requireNonNull;
+
+public class OngoingContextualBranch<C> extends GenericOngoingBranch<OngoingContextualBranch<C>> {
+
+    private final AtomicBoolean asCalled = new AtomicBoolean(false);
+
+    public OngoingContextualBranch(String name, RunnableLeafsMission.Builder builder, Block parent, BranchMode mode) {
+        super(name, builder, parent, mode);
+    }
+
+    public <D extends Consumer<ContextualBranch<C>>> void as(D branchDescription) {
```

Comment:
Why not just
```java
public void as(Consumer<ContextualBranch<C>> branchDescription) {
```
?

Reply:
Indeed! (This is an artifact of some generic-struggle I had ;-)

Reply:
Fixed